tests/README.md

```diff
@@ -182,44 +182,53 @@ expected output. The framework includes the following built-in verify functions:
-   exponentially back off and try again, 7 times by default. The number of retries
-   can be changed by setting the `VERIFY_RETRIES` environment variable.
+   wait a second and try again, up to two minutes by default. The retry behavior
+   can be changed by setting the `VERIFY_TIMEOUT` and `VERIFY_DELAY` environment
+   variables.
 
 1. **`_verify_contains`** `func` `expected`
 
    Runs `func` and compares the output with `expected`. If the output does not
-   contain the substring `expected`, exponentially back off and try again, 7 times
-   by default. The number of retries can be changed by setting the `VERIFY_RETRIES`
-   environment variable.
+   contain the substring `expected`,
+   wait a second and try again, up to two minutes by default. The retry behavior
+   can be changed by setting the `VERIFY_TIMEOUT` and `VERIFY_DELAY` environment
+   variables.
 
 1. **`_verify_not_contains`** `func` `expected`
 
    Runs `func` and compares the output with `expected`. If the command execution fails
    or the output contains the substring `expected`,
-   exponentially back off and try again, 7 times by default. The number of retries
-   can be changed by setting the `VERIFY_RETRIES` environment variable.
+   wait a second and try again, up to two minutes by default. The retry behavior
+   can be changed by setting the `VERIFY_TIMEOUT` and `VERIFY_DELAY` environment
+   variables.
 
 1. **`_verify_elided`** `func` `expected`
 
    Runs `func` and compares the output with `expected`. If the output does not
    contain the lines in `expected` where "..." on a line matches one or more lines
-   containing any text, exponentially back off and try again, 7 times by default.
-   The number of retries can be changed by setting the `VERIFY_RETRIES` environment
-   variable.
+   containing any text,
+   wait a second and try again, up to two minutes by default. The retry behavior
+   can be changed by setting the `VERIFY_TIMEOUT` and `VERIFY_DELAY` environment
+   variables.
 
 1. **`_verify_like`** `func` `expected`
 
    Runs `func` and compares the output with `expected`. If the output is not
-   "like" `expected`, exponentially back off and try again, 7 times by default. The number
-   of retries can be changed by setting the `VERIFY_RETRIES` environment variable.
+   "like" `expected`,
+   wait a second and try again, up to two minutes by default. The retry behavior
+   can be changed by setting the `VERIFY_TIMEOUT` and `VERIFY_DELAY` environment
+   variables.
    Like implies:
 
    - Same number of lines
    - Same number of whitespace-seperated tokens per line
    - Tokens can only differ in the following ways:
 
      1. different elapsed time values (e.g., `30s` is like `5m`)
-     1. different ip values (e.g., `172.21.0.1` is like `10.0.0.31`)
+     1. different ip values (e.g., `172.21.0.1` is like `10.0.0.31`). Allows
+         `<none>` and `<pending>` by default. This can be customized by setting
```

Comment:
Allowing `<pending>` by default seems wrong, since that normally means that it's not ready yet, and that we should really keep retrying until we get a real IP. Usually it's only acceptable if it's "perpetually" pending.

Given the special cases where this is used, do we need two env variables, or would it be good enough to just have one flag `ALLOW_NO_IP` which defaults to false, but when true would allow either `<none` or `<pending>`?

Reply:
I'm actually just maintaining current behavior here. I had to make this change because we weren't retying on `<pending>`. I could change the default, but I'd be concerned other tests might fail .. but maybe it's not a big deal.

Reply:
> Given the special cases where this is used, do we need two env variables, or would it be good enough to just have one flag ALLOW_NO_IP which defaults to false, but when true would allow either <none or <pending>?

The meaning of `none` and `pending` is significantly different. Not sure you can capture that in one flag ... we could try?

Reply:
So the history of this was for setup docs that say `the external IP may be <none> or perpetually <pending>` in which case a nodeport is used for the ingress IP. The framework was trying to handle the case that it's being used on a platform that has no external LB. Matching the `<pending>` then, is only reasonable if there is some kind of `_wait_for` call before the verify call, so retries should never really happen in that case. If, on the other hand, we're using the verify retries as the way to wait for the IP to be ready (as you are doing), the `<pending>` check is wrong, because it is not the final expected state.

So, given all this, I'm thinking maybe should just remove the `<pending>` and `<none>` checks from `_verify_like`. I don't think we do want to test for the nodeport case, and even if we decide to in the future, we will probably do the verify differently anyway. WDYT?

Reply:
ok, let me try that out and see what happens.

Reply:
Looks like there is one test that requires `<pending>`. I've created an issue and captured the details below.